src/status_im2/common/standard_auth/view.cljs

```diff
@@ -0,0 +1,174 @@
+(ns status-im2.common.standard-auth.view
+  (:require [react-native.core :as rn]
+            [quo2.components.numbered-keyboard.numbered-keyboard.view :as numbered-keyboard]
+            [quo2.core :as quo]
+            [status-im2.common.standard-auth.style :as style]
+            [reagent.core :as reagent]
+            [quo2.foundations.colors :as colors]
+            [react-native.fast-image :as fast-image]
+            [status-im2.common.resources :as resources]
+            [utils.i18n :as i18n]
+            [quo2.components.keycard.view :as keycard]
+            [utils.re-frame :as rf]))
+
+(defn enter-keycard-pin-sheet
+  []
+  (let [entered-numbers (reagent/atom [])
+        max-digits 6
+        max-attempt-reached (reagent/atom false)
+
+        handle-number-entry
+        (fn [number]
+          (when (>= (count @entered-numbers) max-digits)
+            (reset! entered-numbers []))
+          (when (= (count @entered-numbers) (dec max-digits))
+            (reset! max-attempt-reached true))
+          (swap! entered-numbers conj number))
+        get-dot-style
+        (fn [idx]
+          (if (<= idx (dec (count @entered-numbers)))
+            (if @max-attempt-reached
+              (assoc style/digit :background-color colors/danger)
+              (assoc style/digit :background-color colors/white))
+            (assoc style/digit :background-color colors/neutral-50)))]
+
+    (fn []
+      (println "entered-numbers" @entered-numbers)
+
+      [rn/view {:style style/container}
+       [rn/view {:style style/inner-container}
+        [quo/text {:accessibility-label :enter-keycard-pin-text :size :heading-2 :weight :semi-bold}
+         (i18n/label :t/enter-keycard-pin)]
+
+        [rn/view {:style style/context-tag}
+         [quo/context-tag
+          {:type    :icon
+           :icon    :i/placeholder
+           :size    24
+           :context (i18n/label :t/alisher-card)}]]
+
+        [rn/view
+         (if @max-attempt-reached
+           {:style style/digits-container :margin-bottom 0}
```

Comment:
I would say it's better to make this style a function and abstract this logic to there 👍

Reply:
@J-Son89 Created a function for it and moved it to style file.